pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -95,17 +98,25 @@ func objectInfoTask(ctxPtr *zedagentContext, triggerInfo <-chan infoForObjectKey
 			case info.ZInfoTypes_ZiApp:
 				// publish application info
 				sub := ctxPtr.getconfigCtx.subAppInstanceStatus
-				if c, err = sub.Get(infoForKeyMessage.objectKey); err == nil {
+				if deleted {
+					PublishAppInfoToZedCloud(ctxPtr, objKey, nil,
+						ctxPtr.assignableAdapters, ctxPtr.iteration, infoDest)
+					ctxPtr.iteration++
+				} else if c, err = sub.Get(objKey); err == nil {
 					appStatus := c.(types.AppInstanceStatus)
-					uuidStr := appStatus.Key()
-					PublishAppInfoToZedCloud(ctxPtr, uuidStr, &appStatus, ctxPtr.assignableAdapters,
-						ctxPtr.iteration, infoDest)
+					PublishAppInfoToZedCloud(ctxPtr, objKey, &appStatus,
+						ctxPtr.assignableAdapters, ctxPtr.iteration, infoDest)
 					ctxPtr.iteration++
```

Comment:
This could be in a separate cleanup PR, but the interation++ seems very ad-hoc now which makes it very hard to check that we always increment it by one for each attempt to send. Here it is done for both the delete and non-delete case, and for ZiNetworkInstance it is only done in the non-delete case. Plus some callers still increment iteration.

Reply:
It is incremented also for [non-delete case of Network Instance](https://github.com/milan-zededa/eve/blob/async-send/pkg/pillar/cmd/zedagent/reportinfo.go#L123), just github diff kept it hidden.

Reply:
I found a few places where the existing code increments ctx*.iteration and one of those was made more visible by these changes. I can do a PR with that cleanup separately from your PR.